types/ioredis/index.d.ts

```diff
@@ -678,192 +594,198 @@ declare namespace IORedis {
+        discard(callback?: Callback<any>): Pipeline;
 
-        sync(callback?: (err: Error, res: any) => void): Pipeline;
+        sync(callback?: Callback<any>): Pipeline;
 
-        flushdb(callback?: (err: Error, res: string) => void): Pipeline;
+        flushdb(callback?: Callback<string>): Pipeline;
 
-        flushall(callback?: (err: Error, res: string) => void): Pipeline;
+        flushall(callback?: Callback<string>): Pipeline;
 
         sort(key: KeyType, ...args: string[]): Pipeline;
 
-        info(callback?: (err: Error, res: string) => void): Pipeline;
-        info(section: string, callback?: (err: Error, res: string) => void): Pipeline;
+        info(callback?: Callback<string>): Pipeline;
+        info(section: string, callback?: Callback<string>): Pipeline;
 
-        time(callback?: (err: Error, res: [string, string]) => void): Pipeline;
+        time(callback?: Callback<[string, string]>): Pipeline;
 
-        monitor(callback?: (err: Error, res: EventEmitter) => void): Pipeline;
+        monitor(callback?: Callback<EventEmitter>): Pipeline;
 
-        ttl(key: KeyType, callback?: (err: Error, res: number) => void): Pipeline;
+        ttl(key: KeyType, callback?: Callback<number>): Pipeline;
 
-        pttl(key: KeyType, callback?: (err: Error, res: number) => void): Pipeline;
+        pttl(key: KeyType, callback?: Callback<number>): Pipeline;
 
-        persist(key: KeyType, callback?: (err: Error, res: BooleanResponse) => void): Pipeline;
+        persist(key: KeyType, callback?: Callback<BooleanResponse>): Pipeline;
 
-        slaveof(host: string, port: number, callback?: (err: Error, res: string) => void): Pipeline;
+        slaveof(host: string, port: number, callback?: Callback<string>): Pipeline;
 
-        debug(...args: ValueType[]): Pipeline;
+        debug(op: 'OBJECT' | 'SEGFAULT', key: string): Pipeline;
 
-        config(...args: ValueType[]): Pipeline;
+        config(op: 'GET', cfg: any[]): Pipeline;
+        config(op: 'REWRITE'): Pipeline;
+        config(op: 'SET', key: string, value: ValueType): Pipeline;
 
-        subscribe(...channels: ValueType[]): Pipeline;
+        subscribe(channels: string | string[], callback?: Callback<number>): Promise<number>;
 
-        unsubscribe(...channels: string[]): Pipeline;
+        unsubscribe(channels: string | string[], callback?: Callback<number>): Promise<number>;
 
-        psubscribe(...patterns: string[]): Pipeline;
+        psubscribe(patterns: string | string[], callback?: Callback<number>): Promise<number>;
 
-        punsubscribe(...patterns: string[]): Pipeline;
+        punsubscribe(patterns: string | string[], callback?: Callback<number>): Promise<number>;
 
-        publish(channel: string, message: string, callback?: (err: Error, res: number) => void): Pipeline;
+        publish(channel: string, message: string, callback?: Callback<number>): Pipeline;
 
         watch(...keys: KeyType[]): Pipeline;
```

Comment:
What about `watch` of the `Pipeline` interface? This function does also have a callback, right?
Same for other methods of the `Pipeline` interface.